shotover-proxy/src/server.rs

```diff
@@ -27,9 +29,114 @@ impl<T: Decoder<Item = Messages, Error = anyhow::Error> + Clone + Send> CodecRea
 pub trait CodecWriteHalf: Encoder<Messages, Error = anyhow::Error> + Clone + Send {}
 impl<T: Encoder<Messages, Error = anyhow::Error> + Clone + Send> CodecWriteHalf for T {}
 
+pub trait CodecErrorFixup {
+    /// An method to fix up an error.
```

Comment:
What does "fix up an error" mean?

Reply:
Ah I see. Is there a scenario where we need to send a message upstream and downstream as part of error handling?